### PR TITLE
Production credentials + Scheduled backup

### PR DESCRIPTION
**Pull request by goldjay**
* firebase credentials based on process environment variables
* Scheduled backup + old backup culling functions

NOTE: I'm going to change the backup to weekly once I verify the scheduler works

NOTE 2: Any recommendations for storing the API info other than an external non-check-in file?

### CODE REVIEW

File: functions/src/index.ts [hunk 1]
```diff
@@ -26,14 +26,21 @@ import {
 
 // You're going to need this file on your local machine.  It's stored in our
 // team's LastPass ServerInfrastructure section.
-const serviceAccount = require("../flows-app-staging-key.json");
+const serviceAccount =
+  !process.env.NODE_ENV || process.env.NODE_ENV === "development"
```
Comment (balderdash) [inline functions/src/index.ts]: Does this work correctly for staging?

File: functions/src/index.ts [hunk 2]
```diff
@@ -57,6 +64,58 @@ type CallResult =
       result: string;
     };
 
+const bucketName = "gs://flows-app-production-backup";
```
Comment (balderdash) [inline functions/src/index.ts]: I'd prefer not to hardcode "production". How about `gs://${serviceAccount.project_id}-backup`?

Same goes for a couple places below where flows-app-production is hardcoded.

File: src/store/corestore.ts [hunk 3]
```diff
@@ -21,16 +21,12 @@ import {
   User,
   UserRole
 } from "../sharedtypes";
+import firebaseConfig from "./firebaseconfig.json";
```
Comment (balderdash) [inline src/store/corestore.ts]: Would this file live in lastpass?

Comment (goldjay) [reply src/store/corestore.ts]: Yes, that's the plan.

File: functions/src/index.ts (hunk 1 above)
Comment (goldjay) [reply functions/src/index.ts]: Turns out no. Looks like the gcloud project id is available in functions so I'll just use that.

**Pull request merged**